packages/push/src/PushRegistration.ts

```diff
@@ -96,22 +97,102 @@ export class PushRegistration {
       "categories": categories
     };
 
-    const instance = axios.create({
+    const http = axios.create({
       baseURL: url,
       timeout: 5000,
       headers: {"Authorization": `Basic ${authToken}`}
     });
 
     return new Promise((resolve, reject) => {
-      return instance.post(PushRegistration.API_PATH, postData)
+      return http.post(PushRegistration.API_PATH, postData)
       .then(
         () => {
+          const storage = window.localStorage;
+          storage.setItem(PushRegistration.TOKEN_KEY, deviceToken);
+
           if (isCordovaAndroid()) {
             this.subscribeToFirebaseTopic(variantId);
             for (const category of categories) {
               this.subscribeToFirebaseTopic(category);
             }
           }
+
+          resolve();
+        }
+      )
+      .catch(reject);
+    });
+  }
+
+  /**
+   * Unregister device for Android or IOS platforms
+   */
+  public unregister(): Promise<void> {
+    if (!window || !window.device) {
+      return Promise.reject(new Error("Registration requires cordova plugin. Verify the " +
+        "@aerogear/cordova-plugin-aerogear-push plugin is installed."));
+    }
+
+    if (!this.pushConfig || !this.pushConfig.config || !this.pushConfig.url) {
+      return Promise.reject(new Error("UPS registration: configuration is invalid"));
+    }
+
+    const storage = window.localStorage;
+    const deviceToken = storage.getItem(PushRegistration.TOKEN_KEY);
+
+    if (!deviceToken) {
+      return Promise.reject(new Error("Device token should not be empty"));
+    }
+
+    let platformConfig;
+    const url = this.pushConfig.url;
```

Comment:
Seems like a strange place to declare this. Would it not make sense to declare it down near line 185?